### PR TITLE
Github Actions CI support

### PR DESCRIPTION
**Pull request by Prajyot-Parab**
Signed-off-by: Prajyot Parab <prajot.parab@ibm.com>

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
```
Comment (yussufsh) [inline .github/workflows/main.yml]: Can we include all the versions of ubuntu and macos. You think we should do docker for centos and debian?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    runs-on: ${{ matrix.os }}
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        ref: 'master'
+
+    - name: Running setup
+      if: ${{ matrix.os != 'windows-latest' }}
```
Comment (yussufsh) [inline .github/workflows/main.yml]: Not sure why we exclude here?

File: .github/workflows/main.yml (hunk 1 above)
Comment (Prajyot-Parab) [reply .github/workflows/main.yml]: Added all supported OS by Github Actions
For Centos and Debian -- let's create a different workflow file

File: .github/workflows/main.yml (hunk 2 above)
Comment (Prajyot-Parab) [reply .github/workflows/main.yml]: the command varies slightly for windows and the rest of the platforms.

**Pull request closed without merging**